### PR TITLE
XPack: core/ccr/Security-cli migration to java-time

### PR DESCRIPTION
**Pull request by pgomulka**
part of the migrating joda time work.
refactoring x-pack plugins usages of joda to java-time
refers #27330

### CODE REVIEW

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/calendars/ScheduledEventTests.java [hunk 1]
```diff
@@ -29,7 +29,8 @@
 public class ScheduledEventTests extends AbstractSerializingTestCase<ScheduledEvent> {
 
     public static ScheduledEvent createScheduledEvent(String calendarId) {
-        ZonedDateTime start = ZonedDateTime.ofInstant(Instant.ofEpochMilli(new DateTime(randomDateTimeZone()).getMillis()), ZoneOffset.UTC);
+        long millis = Clock.system(randomZone()).millis();// TODO we are converting to UTC anyway??
+        ZonedDateTime start = Instant.ofEpochMilli(millis).atZone(ZoneOffset.UTC);
```
Comment (pgomulka) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/calendars/ScheduledEventTests.java]: not sure we need to do soo many convertions here. maybe using just 'ZonedDateTime.now(UTC)` ?

**Review by danielmitterdorfer — APPROVED** [review]
LGTM

File: x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/RollupRequestTranslator.java [hunk 2]
```diff
@@ -220,7 +220,7 @@
                 rolledDateHisto.interval(source.interval());
             }
 
-            String timezone = source.timeZone() == null ? DateTimeZone.UTC.toString() : source.timeZone().toString();
+            String timezone = source.timeZone() == null ? ZoneOffset.UTC.toString() : source.timeZone().toString();
```
Comment (spinscale) [inline x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/RollupRequestTranslator.java]: does this get serialized over the wire somewhere? If so we may need to convert it using `DateUtils`?

File: x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/action/TransportRollupSearchAction.java [hunk 3]
```diff
@@ -288,7 +288,7 @@ static QueryBuilder rewriteQuery(QueryBuilder builder, Set<RollupJobCaps> jobCap
             String fieldName = range.fieldName();
             // Many range queries don't include the timezone because the default is UTC, but the query
             // builder will return null so we need to set it here
-            String timeZone = range.timeZone() == null ? DateTimeZone.UTC.toString() : range.timeZone();
+            String timeZone = range.timeZone() == null ? ZoneOffset.UTC.toString() : range.timeZone();
```
Comment (spinscale) [inline x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/action/TransportRollupSearchAction.java]: same as above

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/calendars/ScheduledEventTests.java (hunk 1 above)
Comment (spinscale) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/calendars/ScheduledEventTests.java]: ack. can you check why we need random timezone here at all?

**Review by spinscale — COMMENTED** [review]
I left few comments to be sure this works. especially when in 'mixed mode' during rolling upgrades (I do not know if anything gets serialized over the write, but it might be worth checking)

File: x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/RollupRequestTranslator.java (hunk 2 above)
Comment (pgomulka) [reply x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/RollupRequestTranslator.java]: this is read in `ValuesSourceAggregationBuilder` so I guess it is already taken care of?
with
```
 if (in.getVersion().before(Version.V_7_0_0)) {
            timeZone = DateUtils.dateTimeZoneToZoneId(in.readOptionalTimeZone());
        } else {
            timeZone = in.readOptionalZoneId();
        }
```

File: x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/action/TransportRollupSearchAction.java (hunk 3 above)
Comment (pgomulka) [reply x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/action/TransportRollupSearchAction.java]: That is read from `RangeQueryBuilder` 
does not seem to be using DateUtils
` timeZone = in.readOptionalZoneId();`
do you think that should be fixed in server?

File: x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/RollupRequestTranslator.java (hunk 2 above)
Comment (spinscale) [reply x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/RollupRequestTranslator.java]: ++

**Comment by dimitris-athanasiou** [comment]
Looks good for the ML changes.

**Review by spinscale — APPROVED** [review]

**Comment by pgomulka** [comment]
this is failing the build because of this change in `RollupRequestTranslator.java`

 ```String timezone = source.timeZone() == null ? DateTimeZone.UTC.toString() : source.timeZone().toString();```
to this
````             String timezone = source.timeZone() == null ? ZoneOffset.UTC.toString() : source.timeZone().toString();```
 This should be fixed in https://github.com/elastic/elasticsearch/pull/36237/files and the string comparison won't be necessary

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci

**Comment by pgomulka** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by polyfractal** [comment]
Re: the rollup bits, I've updated #36237 and will try to get it merged soon

**Pull request merged**